src/lib.rs

```diff
@@ -72,10 +72,8 @@ pub fn run(args: Args) -> Result<()> {
         results.push(uploader.upload_remote_url(remote_url));
     } else if !std::io::stdin().is_terminal() || args.files.contains(&String::from("-")) {
         let mut buffer = Vec::new();
-        let stdin = io::stdin();
-        for bytes in stdin.bytes() {
-            buffer.push(bytes?);
-        }
+        let mut stdin = io::stdin();
+        stdin.read_to_end(&mut buffer)?;
```

Comment:
What happens if we provide a very big file through stdin? I'm assuming it will try to read it in one go and allocate hella memory. Does it make sense to use [`BufReader`](https://doc.rust-lang.org/std/io/struct.BufReader.html) in this case?

Reply:
afaiu `read_to_end` impl already takes a bunch of precautions like reasonable buffering etc. so I don't see any gains

Reply:
alright, I made some tests and it is indeed better than `bytes` approach.